auth/realm/base/src/main/java/org/wildfly/security/auth/realm/JaasSecurityRealm.java

```diff
@@ -60,33 +63,60 @@
  */
 public class JaasSecurityRealm implements SecurityRealm {
 
-    private final String loginConfiguration;
-
+    private static final String DEFAULT_CONFIGURATION_POLICY_TYPE = "JavaLoginConfig";
+    private final URI jaasConfigFilePath;
+    private final String entry;
     private final CallbackHandler handler;
+    private final ClassLoader classLoader;
 
     /**
      * Construct a new instance.
      *
-     * @param loginConfiguration the login configuration name to use
+     * @param jaasConfigFilePath path to JAAS configuration file
+     * @param classLoader        classLoader to use with LoginContext, this class loader must contain LoginModule CallbackHandler classes
      */
-    public JaasSecurityRealm(final String loginConfiguration) {
-        this(loginConfiguration, null);
+    public JaasSecurityRealm(final String jaasConfigFilePath, ClassLoader classLoader) {
+        this(jaasConfigFilePath, null, classLoader, null);
     }
 
     /**
      * Construct a new instance.
      *
-     * @param loginConfiguration the login configuration name to use
-     * @param handler the JAAS callback handler to use
+     * @param jaasConfigFilePath path to JAAS configuration file
+     * @param entry              login configuration file entry
+     * @param classLoader        classLoader to use with LoginContext, this class loader must contain LoginModule CallbackHandler classes
```

Comment:
Would be good to mention in the javadoc for each param whether or not {@code null} is allowed (here and in the other constructors too).